x/tx/signing/aminojson/options.go

```diff
@@ -12,14 +13,38 @@ import (
 
 // getMessageAminoName returns the amino name of a message if it has been set by the `amino.name` option.
 // If the message does not have an amino name, then the function returns false.
-func getMessageAminoName(messageOptions proto.Message) (string, bool) {
+func getMessageAminoName(msg protoreflect.Message) (string, bool) {
+	messageOptions := msg.Descriptor().Options()
 	if proto.HasExtension(messageOptions, amino.E_Name) {
 		name := proto.GetExtension(messageOptions, amino.E_Name)
 		return name.(string), true
 	}
+
 	return "", false
 }
 
+// getMessageAminoName returns the amino name of a message if it has been set by the `amino.name` option.
+// If the message does not have an amino name, then it returns the msg url.
+// If it cannot get the msg url, then it returns false.
+func getMessageAminoNameAny(msg protoreflect.Message) string {
```

Comment:
Could be done after, but it would be beneficial to have a test which assets this behaviour.